bindinfo/bind_cache.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bindinfo
+
+import (
+	"github.com/cznic/mathutil"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/hack"
+	"github.com/pingcap/tidb/util/kvcache"
+	"github.com/pingcap/tidb/util/memory"
+)
+
+// bindCache uses the LRU cache to store the bindRecord.
+// The key of the LRU cache is original sql, the value is a slice of BindRecord.
+// Note: The bindCache is not thread-safe.
+type bindCache struct {
```

Comment:
It seems that some methods of it are not thread-safe. Suggest making all methods thread-safe which can make `bindCache` easier and safer to use and maintain.